### DISCUSSION TITLE
Lab#A CH7 Design optimization - lab1 Step 7: Pipeline the Function

### DISCUSSION THREAD

**Discussion by sssh311318** [bol-edu/HLS-SOC-Discussions#40] [Q&A]
我是照著ug871 toturial 的步驟做的，
以下是我的directive的設定，
![image](https://user-images.githubusercontent.com/59499001/224663263-eca3f82b-f53a-4845-8c86-e434d37735fc.png)
設定好之後 run C synthesis出現以下warning，
![image](https://user-images.githubusercontent.com/59499001/224663359-c054a554-e9c2-4819-85de-7cdde66655d4.png)
並出現 II vilation，
![image](https://user-images.githubusercontent.com/59499001/224664044-16d5c34e-76d9-48d8-a77d-8135702da992.png)
他有提供出現 [HLS 200-885] warning時的解決辦法是使用array partition，
我加上array partition之後也還是出現相同的情況，
所以想請問要如何解決這個II violation，
謝謝。

**Comment by JasonYanggg** [comment]
我在做Ch6有遇過這個問題，我是直接把Pipeline的II設大一點，他就不會有warning了，不過有這個warning好像也沒關係，co-simulation還是會過。
